packages/core/src/app/payment/paymentMethod/CreditCardPaymentMethod.spec.tsx

```diff
@@ -196,16 +196,29 @@ describe('CreditCardPaymentMethod', () => {
         expect(defaultProps.deinitializePayment).toHaveBeenCalled();
     });
 
-    it('renders loading overlay while waiting for method to initialize', () => {
+    it('renders loading overlay while waiting for method to initialize', async () => {
         let component: ReactWrapper;
+        const mockInitializePayment = jest.fn().mockImplementation(() => {
+            return new Promise((resolve) => resolve(true));
+        });
 
         component = mount(<CreditCardPaymentMethodTest {...defaultProps} isInitializing />);
 
         expect(component.find(LoadingOverlay).prop('isLoading')).toBe(true);
 
-        component = mount(<CreditCardPaymentMethodTest {...defaultProps} />);
+        component = mount(
+            <CreditCardPaymentMethodTest
+                {...defaultProps}
+                initializePayment={mockInitializePayment}
+                isInitializing={false}
+            />
+        );
+
+        const loadingOverlay = component.find(LoadingOverlay);
+        setTimeout(() => {
```

Comment:
Thanks for explanation. Than you can try to do something like this. It should be work because after `nextTick` we update component and try checking the loadingOverlay's `isLoading` prop with a new updated value

```suggestion
        await new Promise((resolve) => process.nextTick(resolve));

        component.update();

        const loadingOverlay = component.find(LoadingOverlay);
        
        expect(loadingOverlay.prop('isLoading')).toBe(false);
```

Reply:
fixed